File: 은지/LV2/캐시.py
def solution(cacheSize, cities):
    answer = 0
    cache = []
    for city in cities:
        city = city.upper()
        if cacheSize == 0:
            answer += 5
        else:
            if city in cache:
                answer += 1
                cache.remove(city)
                cache.append(city)
            else:
                answer += 5
                if len(cache) < cacheSize:
                    cache.append(city)
                else:
                    cache.pop(0)
                    cache.append(city)
    return answer

cach = 2
c = ["Jeju", "Pangyo", "NewYork", "newyork"]
print(solution(cach, c))

File: 은지/LV2/캐시.js
function solution(cacheSize, cities) {
  var answer = 0;
  var cache = [];
  for (let city of cities) {
    city = city.toUpperCase()
    if (cacheSize === 0) {
      answer += 5
    } else {
      if (cache.includes(city)) {
        answer += 1
        var i = cache.indexOf(city)
        cache.splice(i, 1)
        cache.push(city)
      } else {
        answer += 5
        if (cache.length < cacheSize) {
          cache.push(city)
        } else {
          cache.shift()
          cache.push(city)
        }
      }
    }
  }
  return answer;
}

cach = 3
c = ["Jeju", "Pangyo", "Seoul", "NewYork", "LA", "Jeju", "Pangyo", "Seoul", "NewYork", "LA"]
console.log(solution(cach, c))